src/input/InputPlugin.hxx

```diff
@@ -62,6 +63,11 @@ struct InputPlugin {
 	 */
 	InputStreamPtr (*open)(const char *uri, Mutex &mutex);
 
+	/**
+	 * Dynamically enumerate supported protocols
+	 */
+	const char* (*enumerate)(void** opaque);
```

Comment:
I don't like this API, because I don't like `void` pointers. And your API isn't properly documented:
- how shall this `opaque` parameter be initialized?
- when are you allowed to call this function? Does it require initializing the plugin first?
- can you leave the enumeration loop early, or does this result in memory leaks?
- can you store an `opaque` pointer and reuse it later? Can you copy it?
- what is the lifetime of the returned pointer? (yes, I'm talking about your global variable hack in the FFmpeg plugin...)
This is error-prone to implement and error-prone to use.

Let's design something which is safe to use. For example:
```
std::set<std::string> (*get_prefixes)();
```
This suffers from the usual C++ bloat, but is easy to understand and is nearly impossible to go wrong.